gemac/management.py

```diff
@@ -0,0 +1,165 @@
+            return addrfiltermode
+        else:
+            return reserved
+
+    @always_seq(host_interface.clk.posedge, reset=None)
+    def readData():
+        if (not host_interface.miimsel) and host_interface.regaddress[9]:
+            regindex = getregisternumber(host_interface.regaddress)
+            if regindex != reserved and host_interface.opcode[1]:  # Read Config
+                host_interface.rddata.next = configregisters[regindex]
+            if regindex == addrtable1 and not host_interface.opcode[1] and \
+                    host_interface.wrdata[23]:  # Address Table Read 0
+                loc = host_interface.wrdata[18:16]
+                addrtablelocation.next = loc
+                host_interface.rddata.next = addresstable[loc][32:0]
+                addrtableread.next = True
+        if not host_interface.miimrdy and mdiodata.done:  # MDIO Read
+            host_interface.rddata.next = mdiodata.rddata[16:]
+        if addrtableread:  # Address Table Read 1
+            addrtableread.next = False
+            host_interface.rddata.next = addresstable[addrtablelocation][48:32]
+
+    @always_seq(host_interface.clk.posedge, reset=None)
+    def writeConfig():
+        if (not host_interface.miimsel) and host_interface.regaddress[9] and \
+                (not host_interface.opcode[1]):
+            regindex = getregisternumber(host_interface.regaddress)
+            if regindex != reserved:
+                configregisters[regindex].next = host_interface.wrdata
+            if regindex == addrtable1 and not host_interface.wrdata[23]:
+                addresstable[host_interface.wrdata[18:16]].next = \
+                    (host_interface.wrdata[16:0] << 32) | \
+                    configregisters[addrtable0]   # Address Table Write
+
+    @always_seq(host_interface.clk.posedge, reset=None)
+    def mdcdriver():
+        clkDiv = configregisters[managementreg][5:]  # + 1 * 2
+        if mdiodata.hostclk_count == clkDiv:
+            mdio_interface.mdc.next = not mdio_interface.mdc
+            mdiodata.hostclk_count.next = 0
+        else:
+            mdiodata.hostclk_count.next = mdiodata.hostclk_count + 1
+
+    @always_seq(host_interface.clk.posedge, reset=None)
+    def mdioinitiate():
+        if host_interface.hostreq and host_interface.miimsel and \
+                host_interface.miimrdy:
+            host_interface.miimrdy.next = False
+            mdio_interface.tri.next = False
+            mdiodata.wrdata.next = \
+                concat(intbv(0b01)[2:], host_interface.opcode[2:],
+                       host_interface.regaddress[10:], intbv(0b10)[2:],
+                       host_interface.wrdata[16:])[32:]
+        if not host_interface.miimrdy:
+            if mdiodata.wrdone:
+                mdio_interface.tri.next = True
+            if mdiodata.done:
+                host_interface.miimrdy.next = True
+
+    @always_seq(mdio_interface.mdc.posedge, reset=None)
```

Comment:
Why use `always_seq` with `reset=None`?   Why not just use `always(clock.posedge)` if the reset is not used?